pkg/share/manager/jsoncs3/jsoncs3.go

```diff
@@ -355,6 +355,13 @@ func (m *Manager) GetShare(ctx context.Context, ref *collaboration.ShareReferenc
 	if share.IsCreatedByUser(s, user) || share.IsGrantedToUser(s, user) {
 		return s, nil
 	}
+
+	shares := m.Cache.ListSpace(s.ResourceId.StorageId, s.ResourceId.SpaceId)
+	if _, ok := shares.Shares[s.GetId().GetOpaqueId()]; ok {
+		// Members of a space are also allowed to list the share.
```

Comment:
Members of a space are not always allowed to list the shares. We need to check the `ListGrants` permission.

Reply:
but the ocs part has a validation step. Maybe that is sufficient.

Reply:
We could also decide that the share manager always returns the shares and the ocs filters it. This is the case for the `in-memory` and `cs3` share managers.

Reply:
> We could also decide that the share manager always returns the shares and the ocs filters it.

I don't like this. The caller of the share service shouldn't receive shares which the current user shouldn't be able to see.

> This is the case for the in-memory, json and cs3 share managers.

The managers are also stating the files to check the permissions. At least that was the case when I implemented that a couple months ago.

Reply:
> Members of a space are not always allowed to list the shares. We need to check the `ListGrants` permission.

@micbar @butonic, I think we should maybe have a short call about this, because this is not how it is implemented right now.

Reply:
call has happened, we decided to add permission checks to the share manager if the calling user is neither `creator` nor `grantee`.